datalad_registry_client/submit_urls.py

```diff
@@ -0,0 +1,93 @@
+import logging
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Iterator
+from typing import List
+
+import requests
+from datalad import cfg
+from datalad.distribution.dataset import Dataset
+from datalad.interface.base import build_doc
+from datalad.interface.base import Interface
+from datalad.interface.results import get_status_dict
+from datalad.interface.utils import eval_results
+from datalad.support.constraints import EnsureNone
+from datalad.support.constraints import EnsureStr
+from datalad.support.param import Parameter
+
+from datalad_registry.utils import url_encode
+
+lgr = logging.getLogger("datalad.registry.submit_urls")
+
+
+@build_doc
+class RegistrySubmitURLs(Interface):
+    """Submit one or more URLs to a DataLad Registry instance."""
+
+    _params_ = {
+        "endpoint": Parameter(
+            args=("--endpoint",),
+            metavar="URL",
+            doc="""DataLad Registry instance to use (no trailing slash).
+            This defaults to the datalad_registry.endpoint option, if set,
+            or http://127.0.0.1:5000/v1 otherwise.""",
+            constraints=EnsureStr() | EnsureNone(),
```

Comment:
I thought to suggest 
```suggestion
            This defaults to the datalad_registry.endpoint option.""",
            constraints=EnsureStr(),
            default=cfg.get("datalad_registry.endpoint", "http://127.0.0.1:5000/v1"),
```
and then would avoid the need for the code in the body of the function, but I do not see us using `cfg.get` in datalad itself in such cases, so may be there is a reason, and thus let's leave as is.

Reply:
note: I also feel that may be later  should in `datalad_registry.__init__` just add a corresponding entry to https://github.com/datalad/datalad/blob/master/datalad/interface/common_cfg.py#L41 thus providing default there, and then use `cfg.obtain`, and thus avoiding need for default value any time we `get` it... do not remember that kitchen ATM to tell to do that for certain